### PR TITLE
update DM config file

### PR DESCRIPTION
**Pull request by kunmingg**
To make cluster-kubeflow.yaml a user-facing config:
* add ability to edit GPU node configs as user request to use multi GPU per pod
* fix CPU_POOL name
* add comments to guide how to customize config
* add default label ```application: 'kubeflow'``` to kubeflow clusters

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/2655)

### CODE REVIEW

File: deployment/gke/deployment_manager_configs/cluster.jinja [hunk 1]
```diff
@@ -159,8 +160,8 @@ resources:
         # Set min cpu platform to ensure AVX2 is supported.
         minCpuPlatform: 'Intel Broadwell'
         accelerators:
-          - acceleratorCount: 1
-            acceleratorType: {{ properties['gpu-type'] }}
+          - acceleratorCount: {{ properties['gpu-number-per-node'] }}
+            acceleratorType: nvidia-tesla-k80
```
Comment (jlewi) [inline deployment/gke/deployment_manager_configs/cluster.jinja]: Why did you remove gpu-type? Users will likely want to configure this.

K80s are pretty old at this point.

Comment (kunmingg) [reply deployment/gke/deployment_manager_configs/cluster.jinja]: Updated

**Comment by kunmingg** [comment]
@sijieamoy
